itunestoND.py

```diff
@@ -28,14 +28,17 @@ def determine_userID(nd_p):
     conn.close()
     return users[0][0]
 
-def update_playstats(d1, id, playcount, playdate, rating=0):
+def update_playstats(d1, id, playcount, playdate, dateadded, rating=0, loved=0):
     d1.setdefault(id, {})
     d1[id].setdefault('play count', 0)
     d1[id].setdefault('play date', datetime.datetime.fromordinal(1))
+    d1[id].setdefault('date added', datetime.datetime.fromordinal(1))
     d1[id]['play count'] += playcount
     d1[id]['rating'] = rating
+    d1[id]['loved'] = loved
 
     if playdate > d1[id]['play date']: d1[id].update({'play date': playdate})
+    if dateadded > d1[id]['date added']: d1[id].update({'date added': dateadded})
```

Comment:
```suggestion
    if dateadded > datetime.datetime.fromordinal(1) and dateadded < d1[id]['date added']: d1[id].update({'date added': dateadded})
```
I think you want to use the earliest date that is defined. All my files had the date I initially imported the files from the disc.